dpnp/dpnp_iface_arraycreation.py

```diff
@@ -1424,13 +1515,107 @@ def fromfile(file, **kwargs):
+        of the items in the file.
+        Default is the default floating point data type for the device where
+        the returned array is allocated.
+    count : int, optional
+        Number of items to read. ``-1`` means all items (i.e., the complete
+        file).
+    sep : str, optional
+        Separator between items if `file` is a text file.
+        Empty ("") separator means the file should be treated as binary.
+        Spaces (" ") in the separator match zero or more whitespace characters.
+        A separator consisting only of spaces must match at least one
+        whitespace.
+    offset : int, optional
+        The offset (in bytes) from the file's current position. Defaults to 0.
+        Only permitted for binary files.
+    device : {None, string, SyclDevice, SyclQueue}, optional
+        An array API concept of device where the output array is created.
+        The `device` can be ``None`` (the default), an OneAPI filter selector
+        string, an instance of :class:`dpctl.SyclDevice` corresponding to
+        a non-partitioned SYCL device, an instance of :class:`dpctl.SyclQueue`,
+        or a `Device` object returned by
+        :obj:`dpnp.dpnp_array.dpnp_array.device` property.
+    usm_type : {"device", "shared", "host"}, optional
+        The type of SYCL USM allocation for the output array.
+        Default is "device".
+    sycl_queue : {None, SyclQueue}, optional
+        A SYCL queue to use for output array allocation and copying.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        A 1-dimensional array created from data in a text or binary file.
+
     Limitations
     -----------
-    Only float64, float32, int64, int32 types are supported.
+    Parameter `like` is supported only with default value ``None``.
+    Otherwise, the function raises `NotImplementedError` exception.
+
+    Notes
+    -----
+    This uses :obj:`numpy.fromfile` and coerces the result to a DPNP array.
+
+    See also
+    --------
+    :obj:`dpnp.frombuffer` : Construct array from the buffer data.
+    :obj:`dpnp.fromiter` : Construct array from an iterable object.
+    :obj:`dpnp.fromstring` : Construct array from the text data in a string.
+    :obj:`dpnp.load` : Load arrays or pickled objects from files.
+    :obj:`dpnp.save` : Save an array to a binary file.
+    :obj:`dpnp.ndarray.tofile` : Write array to a file as text or binary.
+    :obj:`dpnp.loadtxt` : More flexible way of loading data from a text file.
+
+    Examples
+    --------
+    Save the data to a temporary file:
+
+    >>> import tempfile
+    >>> x = tempfile.TemporaryFile()
+    >>> x.write(b"\x00\x01\x02\x03\x04")
+    >>> x.flush()
+    >>> x.seek(0)
+
+    Construct an array:
+
+    >>> import dpnp as np
+    >>> np.fromfile(x, dtype="u1")
+    array([0, 1, 2, 3, 4], dtype=uint8)
+
+    Creating an array on a different device or with a specified usm_type
+
+    >>> x.seek(0)
+    >>> x = np.fromfile(x, dtype="u1") # default case
```

Comment:
`x` gets updated here and using `x.seek(0)` below raises an error

Reply:
renamed the variable